android/src/main/kotlin/com/posthog/posthog_flutter/PosthogFlutterPlugin.kt

```diff
@@ -15,127 +14,89 @@ import io.flutter.plugin.common.MethodChannel.Result
 
 /** PosthogFlutterPlugin */
 class PosthogFlutterPlugin : FlutterPlugin, MethodCallHandler {
-    /// The MethodChannel that will the communication between Flutter and native Android
-    ///
-    /// This local reference serves to register the plugin with the Flutter Engine and unregister it
-    /// when the Flutter Engine is detached from the Activity
     private lateinit var channel: MethodChannel
+    private lateinit var applicationContext: Context
 
     override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
         channel = MethodChannel(flutterPluginBinding.binaryMessenger, "posthog_flutter")
-
-        initPlugin(flutterPluginBinding.applicationContext)
-
+        applicationContext = flutterPluginBinding.applicationContext
+        initPlugin(applicationContext)
         channel.setMethodCallHandler(this)
     }
 
-    private fun initPlugin(applicationContext: Context) {
+    private fun initPlugin(context: Context) {
```

Comment:
@shuttlers-ifiok here.

On start, the plugin tries to initialize using the values in the manifest. Where apiKey is empty, initialization process is aborted.

Leaving the developer to manually call configure.